tests/db/test_database.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import patch, call
+import mysql.connector
+from janitor.db.database import Database
+from janitor.types import DbConnectionDetails
+
+
+test_config = DbConnectionDetails(
+    host="test_host", port=1, db_name="test_dbname", username="test_username", password="test_password"
+)
+
+
+@patch("logging.error")
+def test_given_invalid_connection_when_connecting_to_db_then_check_error_messages_logged(mock_error):
+    with patch("mysql.connector.connect", side_effect=mysql.connector.Error()):
+        Database(test_config)
+        assert mock_error.has_calls(
+            call("MySQL connection failed!"),
+            call(f"Exception on connecting to MySQL database: {mysql.connector.Error()}"),
+        )
+
+
+@patch("logging.error")
+def test_given_invalid_connection_when_closing_connection_then_check_error_message_logged(mock_error):
+    with patch("mysql.connector.cursor") as mock_execute:
```

Comment:
Should `mock_execute` actually be called `mock_cursor`?

Reply:
Yes, you're correct for the below tests. For this test, I've changed this to mock the connection instead as I don't need to mock the cursor. For the below tests, I've changed to `mock_cursor`.